core/src/main/kotlin/org/evomaster/core/output/naming/RestActionTestCaseNamingStrategy.kt

```diff
@@ -161,14 +183,52 @@ open class RestActionTestCaseNamingStrategy(
                 result.add(and)
             }
         }
+
+        val numberQueryParams = getNegativeNumberQueryParams(queryParams)
+        numberQueryParams.forEachIndexed { index, queryParam ->
+            result.add("negative")
+            result.add(queryParam.name)
+            if (index != numberQueryParams.lastIndex) {
+                result.add(and)
+            }
+        }
+
+        val emptyStringQueryParams = getEmptyStringQueryParams(queryParams)
+        emptyStringQueryParams.forEachIndexed { index, queryParam ->
+            result.add("empty")
+            result.add(queryParam.name)
+            if (index != emptyStringQueryParams.lastIndex) {
+                result.add(and)
+            }
+        }
     }
 
+    // TODO: need to check if the BooleanGene is not enclosed in an OptionalGene
     private fun getBooleanQueryParams(queryParams: List<QueryParam>): List<QueryParam> {
-        return queryParams.filter { it.getGeneForQuery() is BooleanGene && (it.getGeneForQuery() as BooleanGene).value }
+        return queryParams.filter {
+            val wrappedGene = getWrappedGene(it)
+            wrappedGene is BooleanGene && wrappedGene.value
```

Comment:
i think this is still problematic... as you are assuming that the target gene, possibly boolean, is directly under Optional. there can be other genes in between. Maybe better to have something like:
```
val b = it.getGeneForQuery().getWrappedGene(Boolean::class.java)
b != null && b.staticCheckIfImpactPhenotype() && b.value
```

Reply:
Hi @arcuri82! Glad you brought this up, I was actually going to ask which way you thought best to extract the correct gene in my next PR.
Tested this and it works great for Boolean and String genes. Number genes, not so much since the check in [Gene](https://github.com/WebFuzzing/EvoMaster/blob/master/core/src/main/kotlin/org/evomaster/core/search/gene/Gene.kt#L305) is performed by class equality and a numeric gene is instantiated as an IntegerGene or any other. The two options I can think of are:
1. Add a `hasWrappedGeneOf` method in `Gene` that handles these type of subclasses. Code would then have something like `gene.hasWrappedGeneOf(NumberGene);   val numberGene = gene as NumberGene`
2. Call `getWrappedGene` with all the subclasses of NumberGene until one returns not null or none do (in which case it was not a numeric gene at all)

What do you think? The second one does not scale much, but I'm not sure on how keen we're on adding functions like in the first option to classes like Gene

Reply:
hi @Pgarrett  I just pushed a change to `getWrappedGene`, but i haven't tested it. you can try it out

Reply:
Thanks @arcuri82, it works like charm! Just removed the `TODO`s